controllers/grid/users/reviewer/form/ThankReviewerForm.inc.php

```diff
@@ -103,39 +99,42 @@ public function readInputData()
      */
     public function execute(...$functionArgs)
     {
+        $request = Application::get()->getRequest();
         $reviewAssignment = $this->getReviewAssignment();
         $reviewerId = $reviewAssignment->getReviewerId();
         $reviewer = Repo::user()->get($reviewerId);
         $submission = Repo::submission()->get($reviewAssignment->getSubmissionId());
+        $contextDao = Application::getContextDAO(); /** @var ContextDAO $contextDao */
+        $context = $contextDao->getById($submission->getData('contextId'));
+        $user = $request->getUser();
 
-        $email = new SubmissionMailTemplate($submission, 'REVIEW_ACK', null, null, null, false);
-
-        $email->addRecipient($reviewer->getEmail(), $reviewer->getFullName());
-        $email->setBody($this->getData('message'));
+        // Create mailable and populate with data
+        $mailable = new ReviewAcknowledgement($context, $submission, $reviewAssignment);
+        $mailable->sender($user)->recipients([$reviewer]);
+        $template = $mailable->getTemplate($context->getId());
+        $mailable->body($this->getData('message'))->subject($template->getLocalizedData('subject'));
 
+        HookRegistry::call('ThankReviewerForm::thankReviewer', [$submission, $reviewAssignment, $mailable]);
```

Comment:
It looks like this hook should only be called if `!$this->getData('skipEmail')`

Reply:
Yes, I saw how it was originally but do we really need to bypass the hook if the email is skipped? Review assignment is updated either way and it looks that the hook should be triggered also.